File: carddemo-modernized/src/main/java/com/carddemo/model/TransactionType.java
package com.carddemo.model;

/**
 * Transaction Type entity - modernized from COBOL copybook CVTRA03Y (TRAN-TYPE-RECORD).
 *
 * Legacy COBOL layout (RECLN 60):
 *   TRAN-TYPE       PIC X(02)   -> typeCode
 *   TRAN-TYPE-DESC  PIC X(50)   -> typeDescription
 *
 * Stored in relational DB via JDBC (mirrors legacy DB2 table).
 */
public class TransactionType {

    private String typeCode;
    private String typeDescription;

    public TransactionType() {}

    public TransactionType(String typeCode, String typeDescription) {
        this.typeCode = typeCode;
        this.typeDescription = typeDescription;
    }

    public String getTypeCode() { return typeCode; }
    public void setTypeCode(String typeCode) { this.typeCode = typeCode; }
    public String getTypeDescription() { return typeDescription; }
    public void setTypeDescription(String typeDescription) { this.typeDescription = typeDescription; }
}

File: carddemo/app/models/transaction_type.py
"""Transaction type model — from CVTRA03Y.cpy / DB2 TRANSACTION_TYPE table.

COBOL record layout (60 bytes):
  05 TRAN-TYPE       PIC X(02)   → String(2) PK
  05 TRAN-TYPE-DESC  PIC X(50)   → String(50)
  05 FILLER          PIC X(08)   → not stored

Managed via CICS programs COTRTLIC/COTRTUPC with DB2 SQL.
"""

from sqlalchemy import String
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base


class TransactionType(Base):
    __tablename__ = "transaction_types"

    tran_type: Mapped[str] = mapped_column(String(2), primary_key=True)
    tran_type_desc: Mapped[str] = mapped_column(String(50), default="")
